### PR TITLE
Rear modular armor

### LINKED ISSUE #879: Cannot assign Modular Armor to be rear facing
Windows 10
Megamek Labs bundled with MegaMek HQ 0.47.16
Java version 15.0.1

Modular Armor, when mounted on the torso locations, must be designated whether it protects the front arc or the rear arc per Tactical Ops. We need the option to mark Modular Armor located in the torsos as rear facing, like weapons,

### PR DESCRIPTION
**Pull request by neoancient**
Include a menu item to change facing of torso-mounted modular armor. This also fixes two other issues I discovered in the process:
1. Alt-right click toggles between forward and rear facing on any equipment whether it makes sense or not.
2. The sprayer and light suction system require a firing arc (TM, p. 247) and should be eligible for rear facing.

I double checked MM and verified that a hit against a torso location with modular armor checks the facing.

Fixes #879

### CODE REVIEW

**Comment by Windchild292** [comment]
Am I just missing how the sprayer and light suction system would have changed or is it not included?

**Comment by neoancient** [comment]
I think it's a GitHub hiccup. It's not showing 0dabe59b0. I'm going to try closing this and reopening.

**Comment by neoancient** [comment]
It's showing now.

File: src/megameklab/com/util/Mech/DropTargetCriticalList.java [hunk 1]
```diff
@@ -123,7 +123,9 @@ public void mousePressed(MouseEvent e) {
 
                 final Mounted mount = getMounted();
                 if ((e.getModifiersEx() & InputEvent.ALT_DOWN_MASK) != 0) {
-                    changeWeaponFacing(!mount.isRearMounted());
+                    if (canRearMount(mount)) {
```
Comment (sixlettervariables) [inline src/megameklab/com/util/Mech/DropTargetCriticalList.java:126]: Does this need the similar `!(getUnit() instanceof BattleArmor)` here as below?

Comment (neoancient) [reply src/megameklab/com/util/Mech/DropTargetCriticalList.java:126]: It won't hurt anything. I suspect there are issues with other unit types, but I'm trying to stick close to the specific mech issue.

**Review by sixlettervariables — APPROVED** [review]

**Review by Windchild292 — APPROVED** [review]

**Pull request merged**